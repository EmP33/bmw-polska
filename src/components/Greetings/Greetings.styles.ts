import styled from "styled-components";
import { motion } from "framer-motion";

export const Container = styled(motion.div)`
  width: 100%;
  height: 100vh;
  position: fixed;
  left: 0;
  top: 0;
  background: rgba(0, 0, 0, 0.9);
  color: #fff;
  z-index: 1000;
  display: grid;
  place-items: center;

  .wrapper {
    text-align: center;
    width: 250px;

    h3 {
      margin-bottom: 0.5rem;
      font-size: 1.5rem;
      font-family: var(--font-heading);
      font-weight: bold;
    }

    p {
      margin-bottom: 4rem;
      font-family: var(--font-secondary);
      font-size: 0.8125rem;
      line-height: 1.25rem;
    }
  }
`;
