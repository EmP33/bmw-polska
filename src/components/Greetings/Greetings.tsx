import React from "react";
import { PrimaryButton } from "../../Components.styles";
import { Container } from "./Greetings.styles";

interface Props {
  handleGreetings: () => void;
}

const containerVariants = {
  hidden: {
    opacity: 0,
  },
  visible: {
    opacity: 1,
    transition: {
      duration: 0.5,
    },
  },
  exit: {
    opacity: 0,
  },
};

const Greetings: React.FC<Props> = ({ handleGreetings }) => {
  return (
    <Container
      variants={containerVariants}
      initial="hidden"
      animate="visible"
      exit="exit"
    >
      <div className="wrapper">
        <h3>DZIĘKUJEMY</h3>
        <p>Skontaktujemy się z Tobą w najbliższym możliwym terminie.</p>
        <PrimaryButton onClick={handleGreetings}>OK</PrimaryButton>
      </div>
    </Container>
  );
};

export default Greetings;
