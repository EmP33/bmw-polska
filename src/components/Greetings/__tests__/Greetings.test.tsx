import Greetings from "../Greetings";
import ReactShallowRenderer from "react-test-renderer/shallow";

describe("<Greetings />", () => {
  it("should render properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Greetings handleGreetings={jest.fn()} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
