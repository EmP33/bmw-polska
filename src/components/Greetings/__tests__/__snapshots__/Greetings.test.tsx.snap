// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greetings /> should render properly 1`] = `
<Styled(Component)
  animate="visible"
  exit="exit"
  initial="hidden"
  variants={
    Object {
      "exit": Object {
        "opacity": 0,
      },
      "hidden": Object {
        "opacity": 0,
      },
      "visible": Object {
        "opacity": 1,
        "transition": Object {
          "duration": 0.5,
        },
      },
    }
  }
>
  <div
    className="wrapper"
  >
    <h3>
      DZIĘKUJEMY
    </h3>
    <p>
      Skontaktujemy się z Tobą w najbliższym możliwym terminie.
    </p>
    <styled.button
      onClick={[MockFunction]}
    >
      OK
    </styled.button>
  </div>
</Styled(Component)>
`;
